crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -573,6 +573,56 @@ impl<'db> Bindings<'db> {
                         );
                     }
 
+                    Some(KnownFunction::Dataclass) => {
+                        if let [init, repr, eq, order, unsafe_hash, frozen, match_args, kw_only, slots, weakref_slot] =
+                            overload.parameter_types()
+                        {
+                            let to_bool = |ty: &Option<Type<'_>>, default: bool| -> bool {
+                                if let Some(Type::BooleanLiteral(value)) = ty {
+                                    *value
+                                } else {
+                                    // TODO: emit a diagnostic if we receive `bool`
```

Comment:
Mypy emits `"init" argument must be a True or False literal` in this case, Pyright just ignores it (falls back to the default value), which is what we also do at the moment. It does not seem particularly important.